File: Games/Uzta/src/main/java/jokatu/game/games/uzta/event/ResourcesUpdatedEventHandler.java
package jokatu.game.games.uzta.event;

import jokatu.game.Game;
import jokatu.game.event.SpecificEventHandler;
import jokatu.game.games.uzta.graph.NodeType;
import jokatu.game.games.uzta.player.UztaPlayer;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

import java.text.MessageFormat;

@Component
public class ResourcesUpdatedEventHandler extends SpecificEventHandler<ResourcesUpdatedEvent> {
	@NotNull
	@Override
	protected Class<ResourcesUpdatedEvent> getEventClass() {
		return ResourcesUpdatedEvent.class;
	}

	@Override
	protected void handleCastEvent(@NotNull Game<?> game, @NotNull ResourcesUpdatedEvent event) {
		UztaPlayer player = event.getPlayer();

		for (NodeType type : NodeType.values()) {
			sender.send(
					MessageFormat.format(
							"/topic/resource.game.{0}.{1}.{2}",
							game.getIdentifier(),
							player.getName(),
							type.toString()
					),
					player.getNumberOfType(type)
			);
		}
	}
}

File: Games/Uzta/src/main/kotlin/jokatu/game/games/uzta/event/ResourcesUpdatedEventHandler.kt
package jokatu.game.games.uzta.event

import jokatu.components.stomp.Topic
import jokatu.game.Game
import jokatu.game.event.SpecificEventHandler
import jokatu.game.games.uzta.graph.NodeType
import jokatu.game.games.uzta.player.UztaPlayer
import org.springframework.stereotype.Component

@Component
class ResourcesUpdatedEventHandler : SpecificEventHandler<ResourcesUpdatedEvent>(ResourcesUpdatedEvent::class) {
	override fun handleCastEvent(game: Game<*>, event: ResourcesUpdatedEvent) {
		val player = event.player

		for (type in NodeType.values()) {
			sender.send(PlayerResources(game, player, type), player.getNumberOfType(type))
		}
	}
}

private class PlayerResources(
		game: Game<*>,
		player: UztaPlayer,
		type: NodeType
): Topic<Int>("resource.game.${game.identifier}.${player.name}.$type")